### PR TITLE
RIA-8116 FTPA tab content update

### PR DESCRIPTION
**Pull request by ThomasKKC**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-8116

### Change description ###
FTPA tab content update

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ x ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantHandler.java [hunk 1]
```diff
@@ -139,12 +151,38 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
-        asylumCase.write(FTPA_APPELLANT_APPLICATION_DATE, dateProvider.now().toString());
+        asylumCase.write(FTPA_APPELLANT_APPLICATION_DATE, currentDate);
 
         asylumCase.write(FTPA_APPELLANT_SUBMITTED, YES);
 
         asylumCase.write(APPEAL_DECISION_AVAILABLE, YesOrNo.YES);
 
+        boolean isDlrmSetAside = featureToggler.getValue("dlrm-setaside-feature-flag", false);
+
+        if (isDlrmSetAside) {
```
Comment (MiguelMartinRojas) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantHandler.java:162]: Maybe we should move this part into a private method

**Review by ikirsanov — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantHandler.java [hunk 2]
```diff
@@ -139,12 +151,47 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
+                newFtpaApplication.setFtpaOutOfTimeExplanation(ftpaAppellantOutOfTimeExplanation);
+            }
+
+            Optional<List<IdValue<FtpaApplications>>> maybeExistingFtpaApplictions =
+                    asylumCase.read(FTPA_LIST);
+            List<IdValue<FtpaApplications>> allApplications =
+                    ftpaAppender.append(newFtpaApplication, maybeExistingFtpaApplictions.orElse(emptyList()));
+
+            asylumCase.write(FTPA_LIST, allApplications);
+            //asylumCase.write(IS_FTPA_LIST_VISIBLE, YesOrNo.YES);
```
[the remark this answers is not in the document]
Comment (ThomasKKC) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantHandler.java:194]: So we still show old data on FTPA tab for now.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaMigrateLegacyDataHandler.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && featureToggler.getValue("dlrm-setaside-feature-flag", false)
+                && Arrays.asList(
+                Event.APPLY_FOR_FTPA_APPELLANT,
+                Event.APPLY_FOR_FTPA_RESPONDENT,
+                Event.RESIDENT_JUDGE_FTPA_DECISION
```
Comment (MiguelMartinRojas) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaMigrateLegacyDataHandler.java:61]: You have to use the new event here.

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by MiguelMartinRojas — APPROVED** [review]

**Pull request merged**